### ISSUE TITLE
cannot login from sddm GUI when sddm is started from GRUB option

### ISSUE THREAD

**Issue by qrp73** [sddm/sddm#1706]
I need to setup two grub menuentry, the first one should boot linux arch into usual text mode console and the second one should boot linux arch into plasma GUI. I installed plasma and it works good when started from command line as **systemctl start sddm**. 

I don't want to use automatic startup for sddm, I need a boot time selection from a grub to choose a text mode or graphics boot.

So, I added the following grub menu entry with a **systemd.unit=sddm.service** option for linux:
```
menuentry "Arch Linux Plasma" {
	load_video
	set gfxpayload=keep
	insmod gzio
	insmod part_msdos
	insmod ext2
	set root='hd0,msdos1'
	echo	'Loading Linux linux ...'
	linux	/boot/vmlinuz-linux root=UUID=<...> rw  loglevel=3 systemd.unit=sddm.service
	echo	'Loading initial ramdisk ...'
	initrd	/boot/initramfs-linux.img
}
```
When I started machine with this option, sddm is started and GUI screen with a prompt for user password appears. But when I type correct password for my user, it fails to login. Also attempt to switch to tty2 with Alt+Ctrl+F2 leads to an empty text mode screen with infinite waiting, so there is no way to switch into console. Alt+Ctrl+Del leads to a proper system reboot. 

I found the following error in **journalctl -u sddm**:
```
Apr 11 13:45:14 matrix sddm-greeter[427]: Reading from "/usr/share/xsessions/plasma.desktop"
Apr 11 13:45:14 matrix sddm[320]: Message received from greeter: Login
Apr 11 13:45:14 matrix sddm[320]: Reading from "/usr/share/xsessions/plasma.desktop"
Apr 11 13:45:14 matrix sddm[320]: Reading from "/usr/share/xsessions/plasma.desktop"
Apr 11 13:45:14 matrix sddm[320]: Session "/usr/share/xsessions/plasma.desktop" selected, command: "/usr/bin/startplasma-x11"
Apr 11 13:45:14 matrix sddm-helper[439]: [PAM] Starting...
Apr 11 13:45:14 matrix sddm-helper[439]: [PAM] Authenticating...
Apr 11 13:45:14 matrix sddm-helper[439]: [PAM] Preparing to converse...
Apr 11 13:45:14 matrix sddm-helper[439]: [PAM] Conversation with 1 messages
Apr 11 13:45:14 matrix sddm-helper[439]: [PAM] authenticate: Authentication failure
Apr 11 13:45:14 matrix sddm-helper[439]: [PAM] returning.
Apr 11 13:45:14 matrix sddm[320]: Authentication error: "\"System is booting up. Unprivileged users are not permitted to log in yet. Please come back later. For technical details, see pam_nologin(8).\""
Apr 11 13:45:14 matrix sddm[320]: Authentication error: "Authentication failure"
Apr 11 13:45:14 matrix sddm-greeter[427]: Message received from daemon: LoginFailed
Apr 11 13:45:14 matrix sddm-greeter[427]: Message received from daemon: LoginFailed
Apr 11 13:45:14 matrix sddm-helper[439]: [PAM] Ended.
Apr 11 13:45:14 matrix sddm[320]: Auth: sddm-helper exited with 1
```

How to fix it?

Also I notice annoying behavior. When I start sddm from a command line with **systemctl start sddm** it shows the following red warning:
```
==== AUTHENTICATING FOR org.freedesktop.systemd1.manage-units ====
Authentication is required to start 'sddm.service'.
Authenticatinng as: <username>
Password: <prompt>
```
if I enter password, sddm starts and everything works ok, but if password is not entered it exit back to the bash and shows the following message:
```
Failed to start sddm.service: Connection timed out.
See system logs and 'systemctl status sddm.service' for details.
polkit-agent-helper-1: pam_authenticate failed: Authentication token manipulation error
```

I'm not sure for what reason there is need for a superuser password to start sddm? Is this correct behavior?
How to solve it to avoid enter superuser password each time when I want to start sddm?

I suspect that this behavior is related with login issue from sddm GUI when sddm is started with option **systemd.unit=sddm.service** in the grub menuentry. Isn't it?

**Comment by Vogtinator** [comment]
You'll have to use `graphical.target`, not `sddm.service`.

**Comment by qrp73** [comment]
> You'll have to use `graphical.target`, not `sddm.service`.

how can I do it? 

Just tried this grub menuentry:
```
menuentry "Arch Linux Plasma" {
	load_video
	set gfxpayload=keep
	insmod gzio
	insmod part_msdos
	insmod ext2
	set root='hd0,msdos1'
	echo	'Loading Linux linux ...'
	linux	/boot/vmlinuz-linux root=UUID=<...> rw  loglevel=3 systemd.unit=graphical.target
	echo	'Loading initial ramdisk ...'
	initrd	/boot/initramfs-linux.img
}
```

But it just starts usual text console...

**Comment by Vogtinator** [comment]
Make sure that `sddm.service` is enabled and part of `graphical.target`.

**Comment by qrp73** [comment]
Thanks.

I solved my issue in the following way:
**systemctl get-default** shows that default target in my system is a graphical.target. First, I executed **systemctl enable sddm**, it switch to autoload kde plasma in graphical mode by default, but now I can add kernel option **systemd.unit=multi-user.target** in the grub menuentry and it allows to load system into a text console without graphical environment. 

So, now I have both modes available from grub menu. That is what I was expected.
